cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -5,10 +5,56 @@ use std::{
 
 use indexmap::{IndexMap, IndexSet};
 
-use super::{Precedence, YaccGrammarError, YaccGrammarErrorKind};
+use super::{parser::YaccParser, Precedence, YaccGrammarError, YaccGrammarErrorKind, YaccKind};
 
 use crate::Span;
 
+pub struct ASTValidation {
+    pub ast: GrammarAST,
+    pub errs: Vec<YaccGrammarError>,
+}
+
+/// Provides access to an ast which may or may not be valid,
```

Comment:
I can't quite parse the second sentence relative to the first (e.g. is a valid AST not kept?).

Reply:
I tried to clarify this entire structures documentation in [1e6c7d3](https://github.com/softdevteam/grmtools/pull/354/commits/1e6c7d3556741f4c9dbad09dce4b5bef2f28e5b8)